packages/flutter_tools/test/general.shard/devfs_test.dart

```diff
@@ -332,6 +333,85 @@ void main() {
       Uri.parse('goodbye'): DevFSFileContent(file),
     }, Uri.parse('/foo/bar/devfs/')), throwsA(isA<DevFSException>()));
   });
+
+  testWithoutContext('test handles request closure hangs', () async {
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    final FakeVmServiceHost fakeVmServiceHost = FakeVmServiceHost(
+      requests: <VmServiceExpectation>[createDevFSRequest],
+    );
+    final HttpClient httpClient = MockHttpClient();
+    final MockHttpClientRequest httpRequest = MockHttpClientRequest();
+    when(httpRequest.headers).thenReturn(MockHttpHeaders());
+    when(httpClient.putUrl(any)).thenAnswer((Invocation invocation) {
+      return Future<HttpClientRequest>.value(httpRequest);
+    });
+    int closeCount = 0;
+    final Completer<MockHttpClientResponse> hanger = Completer<MockHttpClientResponse>();
+    final Completer<MockHttpClientResponse> succeeder = Completer<MockHttpClientResponse>();
+    final List<Completer<MockHttpClientResponse>> closeCompleters =
+      <Completer<MockHttpClientResponse>>[hanger, succeeder];
+    succeeder.complete(MockHttpClientResponse());
+
+    when(httpRequest.close()).thenAnswer((Invocation invocation) async {
+      final Completer<MockHttpClientResponse> completer = closeCompleters[closeCount];
+      closeCount += 1;
+      return await completer.future;
+    });
+    when(httpRequest.abort()).thenAnswer((_) {
+      hanger.completeError(const HttpException('aborted'));
+    });
+    when(httpRequest.done).thenAnswer((_) {
+      if (closeCount == 1) {
+        return hanger.future;
+      } else if (closeCount == 2) {
+        return succeeder.future;
+      } else {
+        // This branch shouldn't happen.
+        assert(false);
```

Comment:
`assert(false)` -> `fail('This branch should not happen')`

Reply:
Done.